app/pages/project/workflow-selection.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import isAdmin from '../../lib/is-admin';
+
+class WorkflowSelection extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      error: null,
+      loadingSelectedWorkflow: true,
+      workflow: null
+    };
+  }
+
+  componentDidMount() {
+    this.getSelectedWorkflow(this.props);
+  }
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps.preferences &&
+      nextProps.preferences.preferences &&
+      nextProps.preferences.preferences.selected_workflow &&
+      this.state.workflow
+    ) {
+      if (!nextState.loadingSelectedWorkflow &&
+        nextProps.preferences.preferences.selected_workflow !== this.state.workflow.id
+      ) {
+        this.getSelectedWorkflow(nextProps);
+      }
+    }
+  }
+
+  getSelectedWorkflow({ project, preferences, user }) {
+    this.setState({ loadingSelectedWorkflow: true });
+    // preference workflow query by admin/owner/collab, then workflow query if in project settings, then user selected workflow, then project owner set workflow, then default workflow
+    // if none of those are set, select random workflow
+    let selectedWorkflowID;
+    let activeFilter = true;
+    if (this.props.location.query &&
+      this.props.location.query.workflow &&
+      this.checkUserRoles(project, user)
+    ) {
+      selectedWorkflowID = this.props.location.query.workflow;
+      activeFilter = false;
+      if (preferences && preferences.preferences.selected_workflow !== selectedWorkflowID) {
+        this.handlePreferencesChange('preferences.selected_workflow', selectedWorkflowID);
+      }
+    } else if (this.props.location.query &&
+      this.props.location.query.workflow &&
+      project.experimentalTools &&
+      project.experimentalTools.indexOf('allow workflow query') > -1
```

Comment:
Lines 49 and 50 should be `project.experimental_tools && project.experimental_tools.indexOf('allow workflow query') > -1`. NfN workflow query loaded is broken otherwise. Tests should be updated to catch this failing case too.

Reply:
Good catch. `project.experimental_tools` is defined in the tests too, I just never wrote a test that checked it.